### PR TITLE
Rust Docs

### PR DESCRIPTION
**Pull request by saolsen**
- [x] Quickstart
- [x] Rust Authorization Library
- [x] Install docs
---

- [ ] Rust API docs (incomplete)
- [x] Rust version of Quickstart
- [x] Rust version of Application Types
- [x] Working with Rust Types
- [ ] Rust version of RBAC
- [ ] Rust version of ABAC
- [ ] Rust version of Context
- [ ] Rust version of Multiple Actor Types
- [ ] Rust version of Resources with Inheritance
- [x] Rust version of REPL docs
- [x] Grep docs to ensure "Rust" is added anywhere we currently enumerate supported languages.
- [x] Automate build of Rust API docs
- [x] Link built Rust API docs on Libraries page
- [ ] Added changelog entry.
- [x] Install instructions
- [x] Rust Library Page

### CODE REVIEW

File: docs/getting-started/policies/application-types.rst [hunk 1]
```diff
@@ -246,6 +273,52 @@ using the :ref:`operator-new` operator if the class has been **registered**:
...
+                }
+
+                fn is_called_alice(&self) -> bool {
+                    self.name == "alice"
+                }
+            }
+
+            oso.register_class(
+               User::get_polar_class_builder()
+                    .set_constructor(User::new)
```
Comment (saolsen) [inline docs/getting-started/policies/application-types.rst:305]: Does this work? I was having trouble using constructors that took arguments without a wrapper lambda but that was a few days ago.

File: docs/more/faq.rst [hunk 2]
```diff
@@ -114,7 +114,7 @@ topic, please see our :doc:`use cases page <use-cases>`.
 What languages and frameworks do you support?
 ---------------------------------------------
 
-We currently support Python, Ruby, Java and Node.js, and are actively working on supporting more languages.
+We currently support Python, Ruby, Java and Node.js, Rust, and are actively working on supporting more languages.
```
Comment (saolsen) [inline docs/more/faq.rst:117]:
```suggestion
We currently support Python, Ruby, Java, Node.js and Rust, and are actively working on supporting more languages.
```

File: docs/using/libraries/rust/index.rst [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+.. warning::
+  Polar does not support methods that mutate strings in place.
+
+
+Vectors
+^^^^^^^
+`Vec<T> <https://doc.rust-lang.org/std/vec/struct.Vec.html>`_ is mapped to Polar :ref:`Lists <lists>`, given that ``T: ToPolar``. 
+
+Currently, no methods on ``Vec`` are exposed to Polar.
```
Comment (saolsen) [inline docs/using/libraries/rust/index.rst:66]: Do we expose any list methods in other languages? If so we should follow up on this.

File: docs/using/libraries/rust/index.rst [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+HashMaps
+^^^^^^^^ 
+
+Rust `HashMaps <https://doc.rust-lang.org/std/collections/struct.HashMap.html>`_ are mapped to Polar :ref:`dictionaries`,
+but require that the ``HashMap`` key is a ``String``:
+
+.. code-block:: polar
+  :caption: :fa:`oso` policy.polar
+
+  allow(actor, action, resource) if actor.roles.project1 = "admin";
```
Comment (saolsen) [inline docs/using/libraries/rust/index.rst:100]: Does this work?

File: languages/rust/oso/examples/lib_guide.rs [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        pub roles: HashMap<String, String>,
+    }
+
+    oso.register_class(User::get_polar_class())?;
+    oso.load_str(r#"allow(actor, action, resource) if actor.roles.project1 = "admin";"#)?;
+
+    let user = User {
+        roles: maplit::hashmap! { "project1".to_string() => "admin".to_string() },
+    };
+    assert!(oso.is_allowed(user, "foo", "bar")?);
```
Comment (saolsen) [inline languages/rust/oso/examples/lib_guide.rs:61]: guess so, cool!

File: docs/getting-started/policies/application-types.rst (hunk 1 above)
Comment (samscott89) [reply docs/getting-started/policies/application-types.rst:305]: Will try!

File: docs/using/libraries/rust/index.rst (hunk 3 above)
Comment (samscott89) [reply docs/using/libraries/rust/index.rst:66]: We do, we expose them all, and have parity tests using them

Comment (saolsen) [reply docs/using/libraries/rust/index.rst:66]: Aight, it goes on the list then, maybe even all the iterator methods 😨

File: docs/getting-started/policies/application-types.rst (hunk 1 above)
Comment (samscott89) [reply docs/getting-started/policies/application-types.rst:305]: Constructor worked, but a different example was incorrect.

File: docs/using/libraries/rust/index.rst (hunk 3 above)
Comment (samscott89) [reply docs/using/libraries/rust/index.rst:66]: Let's just make sure people are able to add their own methods :) we can provide the basics.
Who needs iterator methods when you have Polar? :)

**Review by samscott89 — APPROVED** [review]

**Pull request merged**